### PR TITLE
New POST endpoint for dataset statuses

### PR DESCRIPTION
**Pull request by pawbu**
This POST endpoint will help avoid max query length on GET requests in case there are a lot of datasets in the query.

### CODE REVIEW

File: tests/unit/api/test_metadata_api.py [hunk 1]
```diff
@@ -91,6 +91,28 @@ def test_get_current_data_structure_status(flask_app, mocker):
     assert response.json == MOCKED_DATASTRUCTURE
 
 
+def test_get_current_data_structure_status_as_post(flask_app, mocker):
```
Comment (DanielElisenberg) [inline tests/unit/api/test_metadata_api.py:108]: Could you add an extra test case in this function that takes multiple names ?

File: metadata_service/api/metadata_api.py [hunk 2]
```diff
@@ -34,6 +34,19 @@ def get_data_structure_current_status(query: NameParam):
     return response
 
 
+@metadata_api.post("/metadata/data-structures/status")
```
Comment (DanielElisenberg) [inline metadata_service/api/metadata_api.py:37]: 💯

File: tests/unit/api/test_metadata_api.py [hunk 3]
```diff
@@ -41,6 +41,24 @@
     "releaseTime": 123123,
 }
 
+MOCKED_DATASTRUCTURES = [
```
Comment (DanielElisenberg) [inline tests/unit/api/test_metadata_api.py]: Seems like the correct format for the mocked return would be:
```py
MOCKED_DATASTRUCTURES = {
        "INNTEKT_TJENPEN": {
            "description": "dummy",
            "operation": "ADD",
            "releaseTime": 123123,
        },
        "INNTEKT_BANKINNSK": {
            "description": "dummy",
            "operation": "ADD",
            "releaseTime": 123123,
        },
}
```

[source](https://github.com/statisticsnorway/microdata-metadata-service/blob/801b17edd421a8da795c7702ec5f5db512250caf/metadata_service/domain/metadata.py#L21)

File: tests/unit/api/test_metadata_api.py [hunk 4]
```diff
@@ -95,22 +113,27 @@ def test_get_current_data_structure_status_as_post(flask_app, mocker):
...
-        return_value=MOCKED_DATASTRUCTURE,
+        return_value=MOCKED_DATASTRUCTURES,
     )
     response: Response = flask_app.post(
-        url_for(
-            "metadata_api.get_data_structure_current_status_as_post"),
-        json={"names": "INNTEKT_TJENPEN"},
+        url_for("metadata_api.get_data_structure_current_status_as_post"),
+        json={
+            "names": f"{next(iter(MOCKED_DATASTRUCTURES[0]))},{next(iter(MOCKED_DATASTRUCTURES[1]))}"
```
Comment (DanielElisenberg) [inline tests/unit/api/test_metadata_api.py]: considering the change in the other comment we could do:
```py
"names": ",".join(list(MOCKED_DATASTRUCTURES))
```
or if we want to be more explicit:
```py
"names": ",".join(list(MOCKED_DATASTRUCTURES.keys()))
```

File: tests/unit/api/test_metadata_api.py (hunk 3 above)
Comment (pawbu) [reply tests/unit/api/test_metadata_api.py]: You're right, I fix 🏃

**Review by DanielElisenberg — APPROVED** [review]
:100:

**Pull request merged**